framework/src/dirackernels/ReporterPointSource.C

```diff
@@ -0,0 +1,88 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ReporterPointSource.h"
+#include "MooseUtils.h"
+#include "DelimitedFileReader.h"
+
+registerMooseObject("MooseApp", ReporterPointSource);
+
+defineLegacyParams(ReporterPointSource);
+
+InputParameters
+ReporterPointSource::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+
+  params.addClassDescription("Apply a point load defined by Reporter.");
+
+  params.addRequiredParam<std::string>(
+      "x_coord_name",
+      "reporter x-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<std::string>(
+      "y_coord_name",
+      "reporter y-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<std::string>(
+      "z_coord_name",
+      "reporter z-coordinate name.  This uses the reporter syntax <reporter>/<name>.");
+  params.addRequiredParam<std::string>(
+      "value_name", "reporter value name.  This uses the reporter syntax <reporter>/<name>.");
+
+  return params;
+}
+
+ReporterPointSource::ReporterPointSource(const InputParameters & parameters)
+  : DiracKernel(parameters),
+    ReporterInterface(this),
+    _values(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("value_name")), REPORTER_MODE_REPLICATED)),
+    _x_coord(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("x_coord_name")), REPORTER_MODE_REPLICATED)),
+    _y_coord(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("y_coord_name")), REPORTER_MODE_REPLICATED)),
+    _z_coord(getReporterValueByName<std::vector<Real>>(
+        ReporterName(getParam<std::string>("z_coord_name")), REPORTER_MODE_REPLICATED))
+{
+}
+
+void
+ReporterPointSource::addPoints()
+{
+  if (_values.size() != _x_coord.size() || _values.size() != _y_coord.size() ||
+      _values.size() != _z_coord.size())
+  {
+    std::string errMsg =
+        "The value and coordinate vectors are a different size.  \n"
+        "There must be one value per coordinate.  If the sizes are \n"
+        "zero, the reporter or reporter may not have been initialized with data \n"
+        "before the Dirac Kernel is called.  \n"
+        "Try setting \"execute_on = timestep_begin\" in the reporter being read. \n"
+        "value size = " +
+        std::to_string(_values.size()) + ";  x_coord size = " + std::to_string(_x_coord.size()) +
+        ";  y_coord size = " + std::to_string(_y_coord.size()) +
+        ";  z_coord size = " + std::to_string(_z_coord.size());
+
+    mooseError(errMsg);
```

Comment:
Maybe make this `paramError("value_name", errMsg)`

Reply:
what is the difference?

Reply:
This will point the user to the line in the input file where `value_name` was specified.